### ISSUE TITLE
Should we disable GitHub Actions in forks?

### ISSUE THREAD

**Issue by kbenzie** [oneapi-src/unified-runtime#1292]
Some of our jobs, specifically those running on hardware, are already disabled from running in the GitHub Actions of a fork. However, some jobs are enabled to run in a forks GitHub Actions. This can result in some odd behaviour, e.g. that GitHub Actions run in a fork before a PR against the base repo is made. Or when a PR does exists, duplication of jobs can occur. I've personally disabled the GitHub Actions on my fork of this repo to ignore this issue, perhaps we should default to disabling GitHub Actions from running unless its in the base repo?

**Comment by lukaszstolarczuk** [comment]
It's worth noting it uses runners from your own pool of free GH resources. If anyone feels like it's not a good option, they can disable it manually, like you did.

I, personally, use it to my advantage. If we have a traffic jam on our (oneapi-src) runners, I can check much faster if my changes are good or bad and adjust PR before even CI starts here.

**Comment by kbenzie** [comment]
> I, personally, use it to my advantage. If we have a traffic jam on our (oneapi-src) runners, I can check much faster if my changes are good or bad and adjust PR before even CI starts here.

Do you create a PR on your own fork to do this?

I think the odd behaviour noted by @steffenlarsen (which prompted me creating this ticket) was that some of the GitHub Actions were running when no PR existed in either the base repo or the fork.

**Comment by pbalcer** [comment]
> Do you create a PR on your own fork to do this?

I do the same to make sure my PRs are green on the cpu-only jobs before I create a PR to this repo. There's no need to create PRs on forks, since our workflows also trigger [on push](https://github.com/pbalcer/unified-runtime/blob/main/.github/workflows/cmake.yml#L3).

Fwiw, my vote is for keeping things as they are right now.

**Comment by kbenzie** [comment]
> There's no need to create PRs on forks, since our workflows also trigger [on push](https://github.com/pbalcer/unified-runtime/blob/main/.github/workflows/cmake.yml#L3).

I think this is specifically the unexpected behaviour.

**Comment by steffenlarsen** [comment]
The case I had was when I pushed to a branch on my fork that did not have an open PR to upstream, I would get multiple emails about failing CI just from my own branch (logs are long gone, sadly.) I had made no customizations to my fork at the time, so it came as a bit of a surprise. If it is useful for developers I don't mind keeping it, but I think the question here really should be whether it is best as opt-in or opt-out.

**Comment by lukaszstolarczuk** [comment]
I browsed the web and saw no other solution but to disable workflows via code changes (which, as I wrote previously) is not ideal. I also found information, that (from some time) this should not be an issue for new forks, as the forked repos now do not run any CI workflows, unless explicitly approved.

Anyway, we try to mitigate the obvious CI failures, e.g. by disabling workflows requiring specific HW - if there's something similar to that we should disable it.
